algorithms/linfa-linear/src/glm/mod.rs

```diff
@@ -5,24 +5,32 @@ mod hyperparams;
 mod link;
 
 use crate::error::{LinearError, Result};
-use crate::float::{ArgminParam, Float};
+use crate::float::Float;
 use distribution::TweedieDistribution;
 use hyperparams::TweedieRegressorValidParams;
 use linfa::dataset::AsSingleTargets;
 pub use link::Link;
 
-use argmin::core::{ArgminOp, Executor};
+use argmin::core::{CostFunction, Executor, Gradient, State};
 use argmin::solver::linesearch::MoreThuenteLineSearch;
 use argmin::solver::quasinewton::LBFGS;
+use argmin_math::{ArgminAdd, ArgminDot, ArgminMul, ArgminNorm, ArgminSub};
 use ndarray::{array, concatenate, s};
 use ndarray::{Array, Array1, ArrayBase, ArrayView1, ArrayView2, Axis, Data, Ix2};
-use serde::{Deserialize, Serialize};
 
 use linfa::traits::*;
 use linfa::DatasetBase;
 
 impl<F: Float, D: Data<Elem = F>, T: AsSingleTargets<Elem = F>>
     Fit<ArrayBase<D, Ix2>, T, LinearError<F>> for TweedieRegressorValidParams<F>
+where
+    Array1<F>: ArgminAdd<Array1<F>, Array1<F>>,
+    Array1<F>: ArgminSub<Array1<F>, Array1<F>>,
+    Array1<F>: ArgminMul<Array1<F>, Array1<F>>,
+    F: ArgminMul<Array1<F>, Array1<F>>,
+    Array1<F>: ArgminMul<F, Array1<F>>,
+    Array1<F>: ArgminDot<Array1<F>, F>,
+    Array1<F>: ArgminNorm<F>,
```

Comment:
Are all these new bounds required? I still prefer these bounds over the extra `ArgminParam` code, since it's a net decrease in LOC, but I'd like to minimize the bounds as well.

Reply:
I added the traits which where implemented by `ArgminParam` and didn't check which ones where actually needed. Turns out that `Array1<F>: ArgminMul<Array1<F>, Array1<F>>` could be removed.